lib/common/utils.ts

```diff
@@ -126,41 +126,50 @@ export function patchProperty(obj: any, prop: string) {
     }
     if (target.hasOwnProperty(_prop)) {
       return target[_prop];
-    } else {
+    } else if (originalDescGet) {
       // result will be null when use inline event attribute,
       // such as <button onclick="func();">OK</button>
       // because the onclick function is internal raw uncompiled handler
       // the onclick will be evaluated when first time event was triggered or
       // the property is accessed, https://github.com/angular/zone.js/issues/525
       // so we should use original native get to retrieve the handler
       let value = originalDescGet.apply(this);
-      value = desc.set.apply(this, [value]);
-      if (typeof target['removeAttribute'] === 'function') {
-        target.removeAttribute(prop);
+      if (value) {
+        desc.set.apply(this, [value]);
+        if (typeof target['removeAttribute'] === 'function') {
+          target.removeAttribute(prop);
+        }
+        return value;
       }
-      return value;
     }
+    return null;
   };
 
   Object.defineProperty(obj, prop, desc);
 }
 
 export function patchOnProperties(obj: any, properties: string[]) {
+  const hash: any = {};
   if (properties) {
     for (let i = 0; i < properties.length; i++) {
-      patchProperty(obj, 'on' + properties[i]);
-    }
-  } else {
-    const onProperties = [];
-    for (const prop in obj) {
-      if (prop.substr(0, 2) == 'on') {
-        onProperties.push(prop);
-      }
+      const prop = 'on' + properties[i];
```

Comment:
can we have an explicit list rather than iterating over everything? The issue is that closure compiler creates lots of properties on the window with mangled names, and it is quite easy for them to start with `on`